### DISCUSSION TITLE
Ordering foreign key table by another foreign key

### DISCUSSION THREAD

**Discussion by xadh00m** [PostgREST/postgrest#3065] [Q&A] [answered]
### Description of issue

![film](https://github.com/PostgREST/postgrest/assets/1651999/1b48d6e7-96d0-4c87-a629-4f098bed6087)

I´d like to implement a query which returns the list of all films with all nominations ordered by competitions year. I tried e.g. this:

`/films?select=title,nominations(rank, competitions(name, year))&nominations.order=nominations(competitions(year)).asc`

The error indicates that the nested syntax `nominations(competitions(year))` for an embedded filter is not supported (yet).

Can somebody confirm or correct this please?

Thank you very much!

**Comment by laurenceisla** [comment, marked as answer]
Does it work this way?

```
/films?select=title,nominations(rank,competitions(name, year))&nominations.order=competitions(year).asc
```

The embedded resource [should be called directly](https://postgrest.org/en/stable/references/api/resource_embedding.html#top-level-ordering) without the `nominations(...)` wrapper (it's already specified in `nominations.order=`).

**Comment by xadh00m** [comment]
Hi @laurenceisla!

Thank you for your suggestion. It just worked as you described. Great 👍!

Is it correct that this cannot be nested further? E.g. if I would query for all directors including their films with the same data format/order as described above.

Thank you for you help!

**Reply by laurenceisla** [reply]
Ah, do you mean, if it goes like: `directors?select=films(...)`? If you want to sort the nominations by competitions again, then I think it can still be done with `films.nominations.order=competitions(year).asc`.  The only limitation is that is has to be a `to-One` relationship. So if you want to order, say, films using competitions you won't be able to.

> On [Many-to-One](https://postgrest.org/en/stable/references/api/resource_embedding.html#many-to-one) and [One-to-One](https://postgrest.org/en/stable/references/api/resource_embedding.html#one-to-one) relationships, you can use a column of the “to-one” end to sort the top-level.
